tests/php/tests/includes/test_class.wp-job-manager-usage-tracking-data.php

```diff
@@ -44,4 +106,118 @@ public function test_employers() {
 		$data = WP_Job_Manager_Usage_Tracking_Data::get_usage_data();
 		$this->assertEquals( $employer_count, $data['employers'] );
 	}
+
+	/**
+	 * Expired jobs count.
+	 *
+	 * @since 1.30.0
+	 * @covers WP_Job_Manager_Usage_Tracking_Data::get_usage_data
+	 */
+	public function test_get_usage_data_expired_jobs() {
+		$data = WP_Job_Manager_Usage_Tracking_Data::get_usage_data();
+
+		$this->assertEquals( $this->expired_count, $data['jobs_status_expired'] );
+	}
+
+	/**
+	 * Pending jobs count.
+	 *
+	 * @since 1.30.0
+	 * @covers WP_Job_Manager_Usage_Tracking_Data::get_usage_data
+	 */
+	public function test_get_usage_data_pending_jobs() {
+		$data = WP_Job_Manager_Usage_Tracking_Data::get_usage_data();
+
+		$this->assertEquals( $this->pending_count, $data['jobs_status_pending'] );
+	}
+
+	/**
+	 * Pending payment jobs count.
+	 *
+	 * @since 1.30.0
+	 * @covers WP_Job_Manager_Usage_Tracking_Data::get_usage_data
+	 */
+	public function test_get_usage_data_pending_payment_jobs() {
+		$data = WP_Job_Manager_Usage_Tracking_Data::get_usage_data();
+
+		$this->assertEquals( $this->pending_payment_count, $data['jobs_status_pending_payment'] );
+	}
+
+	/**
+	 * Preview jobs count.
+	 *
+	 * @since 1.30.0
+	 * @covers WP_Job_Manager_Usage_Tracking_Data::get_usage_data
+	 */
+	public function test_get_usage_data_preview_jobs() {
+		$data = WP_Job_Manager_Usage_Tracking_Data::get_usage_data();
+
+		$this->assertEquals( $this->preview_count, $data['jobs_status_preview'] );
+	}
+
+	/**
+	 * Published jobs count.
+	 *
+	 * @since 1.30.0
+	 * @covers WP_Job_Manager_Usage_Tracking_Data::get_usage_data
+	 */
+	public function test_get_usage_data_publish_jobs() {
+		$data = WP_Job_Manager_Usage_Tracking_Data::get_usage_data();
+
+		$this->assertEquals( $this->publish_count, $data['jobs_status_publish'] );
+	}
+
+	/**
+	 * Jobs with a company logo count.
+	 *
+	 * @since 1.30.0
+	 * @covers WP_Job_Manager_Usage_Tracking_Data::get_usage_data
+	 * @covers WP_Job_Manager_Usage_Tracking_Data::get_company_logo_count
+	 */
+	public function test_get_company_logo_count() {
+		// Create some media attachments.
+		$media = $this->factory->attachment->create_many(
+			6, array(
+				'post_type'   => 'job_listing',
+				'post_status' => 'publish',
+			)
+		);
+
+		// Add logos to some listings with varying statuses.
+		add_post_meta( $this->listings[0], '_thumbnail_id', $media[0] );
```

Comment:
It's not very clear here which jobs the attachments are being added to. It also seems brittle. Will this break if we change how many jobs of various statuses are being generated?

Reply:
> It's not very clear here which jobs the attachments are being added to.

Fair point. I got lazy. 😛  Fixed in https://github.com/Automattic/WP-Job-Manager/pull/1331/commits/23b6b0407565f7d84671a7f61d7d17c7551e218c.

> Will this break if we change how many jobs of various statuses are being generated?

Yes, it's all isolated in the `create_job_listings` function though. This seems acceptable to me as the same thing would happen if you defined it inside the test case itself and then decided to change the number of jobs to create.